src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -432,30 +432,21 @@ class ResourceService(
   def deleteResource(resource: FullyQualifiedResourceId, samRequestContext: SamRequestContext): IO[Unit] =
     for {
       _ <- checkNoChildren(resource, samRequestContext)
-      _ <- checkNoPoliciesInUse(resource, samRequestContext)
 
       // remove from cloud first so a failure there does not leave sam in a bad state
       _ <- cloudDeletePolicies(resource, samRequestContext)
       _ <- deleteActionManagedIdentitiesForResource(resource, samRequestContext)
 
       // leave a tomb stone if the resource type does not allow reuse
       leaveTombStone = !resourceTypes(resource.resourceTypeName).reuseIds
+      affectedPolicies <- accessPolicyDAO.findAffectedPolicyGroups(resource, samRequestContext) // New method
+      _ <- affectedPolicies.traverse { case (policyToUpdate, policyToRemove) => removeSubjectFromPolicy(policyToUpdate, policyToRemove, samRequestContext) }
+      _ <- cloudSyncPolicies(affectedPolicies, samRequestContext) // TODO: use this on affected policies
```

Comment:
if you are using `removeSubjectFromPolicy`, do you still need the call to `cloudSyncPolicies`? Does `removeSubjectFromPolicy` handle the syncing internally?

Reply:
Not sure, actually.  Maybe @dvoet knows?

Reply:
you don't need to call `cloudSyncPolicies`